lib/main.dart

```diff
@@ -1,18 +1,49 @@
-import 'src/card_list.dart';
+import 'package:first_app/src/data/datasource/remote/card_api_provider.dart';
+import 'package:first_app/src/domain/repository/repository_card.dart';
+import 'package:first_app/src/presentation/bloc/card_bloc.dart';
 import 'package:flutter/material.dart';
-import 'src/card_constants.dart';
+import 'package:http/http.dart';
+import 'src/core/util/card_constants.dart';
+import 'src/presentation/view/card_list.dart';
 
 void main() => runApp(const HearthstoneCardApp());
 
-class HearthstoneCardApp extends StatelessWidget {
+class HearthstoneCardApp extends StatefulWidget {
   const HearthstoneCardApp({super.key});
 
+  @override
+  State<HearthstoneCardApp> createState() => _HearthstoneCardAppState();
+}
+
+class _HearthstoneCardAppState extends State<HearthstoneCardApp> {
+  static final Client client = Client();
+  static final CardApiProvider cardApi = CardApiProvider(client: client);
+  static final RepositoryCard repositoryCard =
+      RepositoryCard(client: client, cardsApi: cardApi);
+  static CardBloc cardBloc = CardBloc(repositoryCard: repositoryCard);
+
+  @override
+  void initState() {
+    super.initState();
+    cardBloc = CardBloc(
+      repositoryCard: RepositoryCard(
+        cardsApi: CardApiProvider(
+          client: Client(),
+        ),
+        client: Client(),
```

Comment:
You already create a cardApiProvider and a client in lines 19-20, why you recreate this componentes again here?

Reply:
For no more apparent reason than a mistake, fixed.